FirebaseTestProject/Models/Post.swift

```diff
@@ -7,24 +7,72 @@
 
 import Foundation
 import FirebaseFirestore
+import SwiftUI
 
 struct Post: FirebaseConvertable {
     let title: String
     let author: String
+    let authorid: UUID
     let text: String
     let id: UUID
     let timestamp: Date
+    var isFavorite: Bool = false
     
     init(title: String, text: String, author: String) {
         self.title = title
-        self.author = author
+        self.author = author == "" ? "Brad" : author
+        let userid = UserDefaults.standard.value(forKey: "userid") != nil ? UUID(uuidString: UserDefaults.standard.value(forKey: "userid") as! String) : UUID(uuidString: "00854E9E-8468-421D-8AA2-605D8E6C61D9")
+        self.authorid = userid!
         self.text = text
         self.id = UUID()
         self.timestamp = Date()
     }
+    
+    private enum CodingKeys: String, CodingKey {
+        case title
+        case author
+        case authorid
+        case text
+        case id
+        case timestamp
+    }
+    
+    //Custom Decoder for `isFavorite` since we don't set it using th Firebase.posts collections
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        title = try container.decode(String.self, forKey: .title)
+        author = try container.decode(String.self, forKey: .author)
+        text = try container.decode(String.self, forKey: .text)
+        id = try container.decode(UUID.self, forKey: .id)
+        timestamp = try container.decode(Date.self, forKey: .timestamp)
+        authorid = try container.decode(UUID.self, forKey: .authorid)
+    }
+    
+    func postSearch(_ string:String) -> Bool {
+        let lowercaseString = string.lowercased()
+        let lowercaseTitle = title.lowercased()
+        let lowercaseAuthor = author.lowercased()
+        let lowercaseText = text.lowercased()
+        let lowercaseDate = DateFormatter.postFormat(date: timestamp).lowercased()
```

Comment:
These probably don't each need to have a separate property.  Maybe a map could make it a bit more readable?